docs/docs/porting-create-react-app-to-gatsby.md

```diff
@@ -0,0 +1,263 @@
+
+---
+
+## What do you get by changing to Gatsby?
+
+Both Create React App and Gatsby use React and allow a faster setup, especially in setting up the initial configuration of a site, but their are differences.
+
+### Performance optimizations
+
+A lot of automatic optimization upgrades like automatic code splitting and preloading of data come out of the box with Gatsby to improve site speed and performance audit scores. There are improvements like progressive web app support that are part of both projects, but Gatsby offers even more specific performance optimizations.
+
+A few examples:
+
+- `gatsby-link` uses an [Intersection Observer to preload linked pages](/blog/2019-04-02-behind-the-scenes-what-makes-gatsby-great/#intersectionobserver) when they appear in the viewport, making them feel like they load _instantly_
+- `gatsby-image` will create optimized versions of your images in different sizes, loading a lighter version of an image and replacing it with a higher res version when loading has finished, check out [the demo](https://using-gatsby-image.gatsbyjs.org/) to see for yourself
+
+These and more work together to make your Gatsby site fast by default.
+
+### Expansive ecosystem
+
+Gatsby still works with all the `react` packages running in a Create React App project, but adds the ability to leverage [hundreds of other Gatsby plugins](/plugins) to drop in additional features for things like adding SEO, responsive images, RSS feeds, or much more.
+
+### Unified GraphQL data layer
+
+Plugins can also pull in data from any number of sources like APIs, CMSes, or the filesysten. That data is combined into a unified data layer that you can query with GraphQL throughout your app.
+
+This data layer simplifies the process of pulling data from different sources and providing them in your pages and components.
+
+> **Note**: GraphQL isn't required for managing data in a Gatsby app, feel free to look over the guide on [using Gatsby without GraphQL](/docs/using-gatsby-without-graphql/) as well
+
+---
+
+## What changes need to be made?
+
+In order to transition your codebase over to using Gatsby, a few things need to be taken care of to account for the differences between how the projects are set up.
+
+In order to use Gatsby, you have to install it:
+
+```shell
+npm install --save gatsby
+```
+
+> **Note**: rather than using the `gatsby new` command like you would initializing a new Gatsby site, this will install Gatsby as a dependency in your project
+
+After installation, the key things that need to change are:
+
+1. protecting any calls to browser based APIs (if there are any)
```

Comment:
What do you mean by "protecting"? Perhaps "ensuring any calls to browser-based APIs still work", or "supporting calls to browser-based APIs"?

Reply:
Yeah protecting is a little vague, the debugging html builds page says checking if things like window and and document are defined and I was searching for my own term for it, I like your first suggestion.